### PR TITLE
Apply CalcMergeRule also for physical opt phase

### PR DESCRIPTION
**Pull request by Fly-Style**
Recently, I notice that sometimes we have a final query plan which contains two physical `Calc` consequently, after the physical opt phase. This change open an optimization for such queries by merging two physical `Calc`.

### CODE REVIEW

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/PhysicalRules.java [hunk 1]
```diff
@@ -30,6 +31,7 @@ public static RuleSet getRuleSet() {
         return RuleSets.ofList(
                 // Filter rules
                 CalcPhysicalRule.INSTANCE,
+                CalcMergeRule.INSTANCE,
```
Comment (viliam-durina) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/PhysicalRules.java:34]: 1. the rule is in `opt.logical` package, is that correct?
2. why don't we use calcite's `CoreRules.CALC_MERGE`?

Comment (Fly-Style) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/PhysicalRules.java:34]: 1. I'll move it.
2. As Ozerov mentioned [here](https://github.com/hazelcast/hazelcast/issues/17058#issue-631615254), we needed to implement custom `Calc`-`Calc` rule, what was done.

**Review by ivanthescientist — APPROVED** [review]

**Review by krzysztofslusarski — APPROVED** [review]

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/common/CalcMergeRule.java [hunk 2]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package com.hazelcast.jet.sql.impl.opt.logical;
+package com.hazelcast.jet.sql.impl.opt.common;
```
Comment (viliam-durina) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/common/CalcMergeRule.java:17]: We must attribute Flink for using their code.

**Review by viliam-durina — CHANGES_REQUESTED** [review]

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/PhysicalRules.java (hunk 1 above)
Comment (viliam-durina) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/PhysicalRules.java:34]: 2. Still, I'd like t see:
  - why we need a different rule than the one calcite has
  - why we need it in the physical phase

**Pull request closed without merging**